cmd/yggctl/actions.go

```diff
@@ -42,6 +56,64 @@ func generateControlMessageAction(c *cli.Context) error {
 	return nil
 }
 
+func messageJournalAction(ctx *cli.Context) error {
+	var conn *dbus.Conn
+	var err error
+
+	if os.Getenv("DBUS_SESSION_BUS_ADDRESS") != "" {
+		conn, err = dbus.ConnectSessionBus()
+		if err != nil {
+			return cli.Exit(fmt.Errorf("cannot connect to session bus: %w", err), 1)
+		}
+	} else {
+		conn, err = dbus.ConnectSystemBus()
+		if err != nil {
+			return cli.Exit(fmt.Errorf("cannot connect to system bus: %w", err), 1)
+		}
+	}
+
+	var journalEntries []map[string]string
+	args := []interface{}{
+		ctx.Uint("truncate-message"),
+		ctx.String("message-id"),
+		ctx.String("worker"),
+		ctx.String("from"),
+		ctx.String("to"),
+		ctx.Bool("persistent"),
+	}
+	obj := conn.Object("com.redhat.Yggdrasil1", "/com/redhat/Yggdrasil1")
+	if err := obj.Call("com.redhat.Yggdrasil1.MessageJournal", dbus.Flags(0), args...).Store(&journalEntries); err != nil {
+		return cli.Exit(fmt.Errorf("cannot list message journal entries: %v", err), 1)
+	}
+
+	if len(journalEntries) == 0 {
+		fmt.Println("No journal entries found.")
+		return nil
+	}
+
+	writer := tabwriter.NewWriter(os.Stdout, 4, 4, 2, ' ', 0)
+	fmt.Fprint(
+		writer,
+		"MESSAGE #\tMESSAGE ID\tSENT\tWORKER NAME\tRESPONSE TO\tWORKER EVENT\tWORKER MESSAGE\n",
+	)
+	for idx, entry := range journalEntries {
+		fmt.Fprintf(
+			writer,
+			"%d\t%s\t%s\t%s\t%s\t%v\t%s\n",
+			idx,
+			entry["message_id"],
+			entry["sent"],
+			entry["worker_name"],
+			entry["response_to"],
+			entry["worker_event"],
+			entry["worker_message"],
+		)
```

Comment:
The worker events being emitted don't contain a `response_to` field, so I instead added an entry from the dispatcher when a message is being dispatched on the worker. In the case of the `echo` worker the `response_to` field isn't being utilized.

Although I think it would be fine if the worker events emitted the "response_to" field as well to prevent this case... any thoughts @subpop ?

Reply:
Hm. I suppose it's harmless to include the response_to field with events. The worker event already includes the message_id, so a second UUID is reasonable enough.